acapy_agent/protocols/present_proof/v2_0/formats/anoncreds/handler.py

```diff
@@ -292,15 +303,21 @@ def _check_proof_vs_proposal():
                         f"Proposed request predicate on {req_name} not in presentation"
                     )
 
+                sub_proof_index = attr_spec["sub_proof_index"]
```

Comment:
This line introduces a bug (its already defined correctly in line 285

Reply:
Fixing. Ya this function should really be refactored later.